### PR TITLE
Refactored to use Ampersand.js and webpack

### PR DESCRIPTION
**Pull request by nchapman**
Refactored the _hack n' slash_ app that we put together into a more maintainable app using [Ampersand.js](http://ampersandjs.com/), [webpack](http://webpack.github.io/), and [Babel](https://babeljs.io/) (among other things).

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  module: {
+    loaders: [
+      {
+        test: /\.js$/,
+        exclude: /(node_modules|bower_components)/,
+        loader: 'babel'
+      },
+      { test: /\.html$/, loader: 'mustache' },
+      { test: /\.scss$/, loader: 'style!css!sass?sourceMap' },
+      { test: /\.png$/, loader: "url-loader?limit=100000" }
```
Comment (nchapman) [inline webpack.config.js]: Replace double quotes.

File: package.json [hunk 2]
```diff
@@ -1,28 +1,39 @@
...
   "scripts": {
+    "build": "./node_modules/.bin/webpack --optimize-minimize --optimize-dedupe",
+    "start": "./node_modules/.bin/webpack-dev-server --content-base dist/",
     "test": "echo \"Error: no test specified\" && exit 1"
   },
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/mozilla/universal-search-content"
+  "author": "Mozilla",
+  "license": "MPL 2.0",
```
Comment (pdehaan) [inline package.json]: Should be "MPL-2.0", if we want to be SPDX compliant (or newer npm's will whine).

File: package.json [hunk 3]
```diff
@@ -1,28 +1,39 @@
 {
   "name": "universal-search-content",
-  "version": "1.0.0",
-  "description": "the iframe that we put inside the autocomplete dropdown",
+  "version": "0.0.1",
+  "description": "",
   "main": "index.js",
   "scripts": {
+    "build": "./node_modules/.bin/webpack --optimize-minimize --optimize-dedupe",
+    "start": "./node_modules/.bin/webpack-dev-server --content-base dist/",
```
Comment (pdehaan) [inline package.json]: IIRC, npm adds ./node_modules/.bin/ to the $PATH, so you shouldn't need these prefixes in npm scripts.

File: package.json (hunk 2 above)
Comment (nchapman) [inline package.json]: Nice thanks!

File: package.json (hunk 3 above)
Comment (lmorchard) [inline package.json]: I don't think it does, or at least it doesn't work reliably. I had to add `./node_modules/bin` to my `$PATH` in my `.zshrc`

Comment (nchapman) [inline package.json]: :+1:

**Pull request merged**